### ISSUE TITLE
Changing locale and calandar

### ISSUE THREAD

**Issue by ImanMh** [airbnb/react-dates#259]
Since react-dates is based on moment and moment supports many locales and calendars, there should be an easy way to change calendar. 
First I want to know what's the proper way of switching locale before react-dates renders the component?
Second I want to know what's the proper way of switching calendar? This one is a little bit more complex. Suppose I want to add [moment-jalaali](https://github.com/jalaali/moment-jalaali). I have to import it and pass the instance to the react-dates but since react-dates uses it's own imported instance of moment, it will not see moment-jalaali.

**Comment by ljharb** [comment]
`moment` is a singleton (that's why it's a peerDependency) - when you add a locale plugin, react-dates' `moment` gets it too.

**Comment by BrunoAl** [comment]
I am having the same problem, I am trying to change the locale to pt-Br (Brazilian Portuguese) but I can't figure out how.

**Comment by BrunoQuaresma** [comment]
I am have the same issue.

**Comment by majapw** [comment]
As @ljharb mentioned, `moment` is a peer dependency of `react-dates` not a regular dependency (https://github.com/airbnb/react-dates/blob/master/package.json#L103). This means that when you `npm install react-dates`, you don't get a separate instance of `moment` in that tree. Whatever instance of `moment` you have in your app where you require `react-dates` is the same instance `react-dates will use.

You can then set the date before you render the component in the way we do in the `non-english locale` example here: https://github.com/airbnb/react-dates/blob/master/stories/DateRangePicker.js#L177:L191

**Comment by majapw** [comment]
@BrunoAl `moment.locale('pt-br')` is probably what you need. I think case is important.

**Comment by BrunoQuaresma** [comment]
@majapw thanks!!!!!!!!

**Comment by asliwinski** [comment]
I arrived here by googling "react dates locale". I think the above should be mentioned in the readme.

**Comment by ljharb** [comment]
@asliwinski a PR would be appreciated :-) that said, this is part of moment.js, so we wouldn't want much more than a sentence and a link to moment's docs.

**Comment by KnockNow** [comment]
@majapw yes it works but only when mounting the component. If my locale is updated after, it doesn't work. 
Why not make a local setting ? Where can you be a solution ?

**Comment by majapw** [comment]
@KnockNow I get the frustration but I don't feel like it is appropriate for `react-dates` to take care of what is essentially a global setting outside of the package. Because moment locale affects everything in your app that might use dates, I think it makes sense for the consumer to set it themselves in the appropriate place.

**Comment by KnockNow** [comment]
@majapw Sorry, I do not understand your logic. If this is the overall setting of the project, why not connect it to your lib? I clone your repo and I try, it works. For a calendar that is called "internationalizable"...

**Comment by tranan89** [comment]
@majapw You should make locale a prop or let us pass in a local moment instance. Changing the global moment locale isn't good enough for a an international app that server side renders with multiple languages. Now we probably have to change to another datepicker...

**Comment by krcaltomas99** [comment]
If you want to change locales on the fly, you have to create an instance from moment js and then if you change the language, you have to call the change again. This will probably change the date if you already selected one, but I'm sure you can fix that very easily 😇

Working Example

`
...
const [date, setDate] = useState(moment());

useEffect(() => {
	moment.locale(i18n.language);
	setDate(moment());
}, [i18n.language]);
..
`

**Comment by espretto** [comment]
Hello, just for anyone looking for a quick solution to trick react-dates into adopting a given locale on every (server-side-)render, use this hack: https://github.com/airbnb/react-dates/issues/327#issuecomment-414840540

**Comment by kumphol** [comment]
I have tried to fixed this locale problem for a couple hours. Then I asked chatGPT and I found something interesting.
chatGPT is add this line when import

`import 'moment/locale/th';`

At first I import it into single page that use dateRangePicker and it worked fine but other pages didn't change.

Then I put `import 'moment/locale/th'` at the top file on my project, in my case I use nextjs then I import it at _app.page.tsx and when I change language I do `moment.locale('th' or 'your locale')` and every page worked like a hell. 

Hope this work for you too.
